File: manga_frontend/src/types/history.ts
import type { UUID } from "./common";

export interface HistoryEntry {
  HistoryId: UUID;
  MangaId: UUID;
  ChapterId: UUID;
  LastPageRead?: number | null;
  ReadAt?: string | null;
  manga_title?: string | null;
  chapter_number?: string | null;
  cover_url?: string | null;
}

export interface HistoryGroup {
  label: string;
  items: HistoryEntry[];
}

export interface GroupedHistoryResponse {
  groups: HistoryGroup[];
}

export interface HistoryCreatePayload {
  MangaId: UUID;
  ChapterId: UUID;
  LastPageRead?: number | null;
}

export interface ContinueReadingResponse {
  chapter_id?: UUID | null;
  last_page?: number | null;
}

File: manga_backend/app/schemas/history.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class HistoryCreate(BaseModel):
    MangaId: uuid.UUID
    ChapterId: uuid.UUID
    LastPageRead: int | None = None


class HistoryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    HistoryId: uuid.UUID
    MangaId: uuid.UUID
    ChapterId: uuid.UUID
    LastPageRead: int | None = None
    ReadAt: datetime | None = None
    manga_title: str | None = None
    chapter_number: str | None = None
    cover_url: str | None = None


class HistoryGroup(BaseModel):
    label: str
    items: list[HistoryResponse] = []
